spec/diffux_ci_spec.rb

```diff
@@ -523,4 +523,65 @@ def snapshot_file_exists?(description, size, file_name)
         .to eq(false)
     end
   end
+
+  describe 'when additional files are served from public directories' do
+    before do
+      @tmp_pub_dir = File.join(@tmp_dir, 'public')
+      Dir.mkdir(@tmp_pub_dir)
+
+      File.open(File.join(@tmp_pub_dir, 'picture.gif'), 'wb') do |f|
+        f.write(Base64.decode64('R0lGODlhAQABAIABAP///wAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=='))
+      end
+    end
+
+    let(:config) do
+      {
+        'source_files' => ['examples.js'],
+        'public_directories' => ['public']
+      }
+    end
+
+    let(:examples_js) { <<-EOS }
+      diffux.define('img', function() {
+        return new Promise( function(resolve, reject) {
+          var image = new Image();
+          image.onload = function() {
+            resolve(image);
+          };
+          image.onerror = function() {
+            reject(new Error('image not found'));
```

Comment:
Our approach here is a little unusual, so it is probably worth documenting why we are doing this weird stuff with a comment.